src/main/java/nl/tudelft/dnainator/ui/controllers/FileOpenController.java

```diff
@@ -0,0 +1,175 @@
+
+	@FXML private FileOpenPane fileOpenPane;
+	@FXML private TextField nodeField;
+	@FXML private TextField edgeField;
+	@FXML private TextField newickField;
+
+	private GraphLoadService graphLoadService;
+	private NewickLoadService newickLoadService;
+	private FileChooser fileChooser;
+	private ProgressDialog progressDialog;
+	private ObjectProperty<PhylogeneticTree> treeProperty;
+
+	/*
+	 * Sets up the services, filechooser and treeproperty.
+	 */
+	@FXML
+	private void initialize() {
+		fileChooser = new FileChooser();
+		treeProperty = new SimpleObjectProperty<>(this, "tree");
+
+		graphLoadService = new GraphLoadService();
+
+		graphLoadService.setOnFailed(e ->
+				new ExceptionDialog(fileOpenPane.getParent(), graphLoadService.getException(),
+						"Error loading graph files!"));
+		graphLoadService.setOnRunning(e -> progressDialog.show());
+		graphLoadService.setOnSucceeded(e -> progressDialog.close());
+
+		newickLoadService = new NewickLoadService();
+		newickLoadService.setOnFailed(e ->
+				new ExceptionDialog(fileOpenPane.getParent(), newickLoadService.getException(),
+						"Error loading newick file!"));
+		newickLoadService.setOnSucceeded(e -> treeProperty.setValue(
+				new PhylogeneticTree(newickLoadService.getValue())
+		));
+	}
+
+	/*
+	 * If the node textfield is clicked, open the filechooser and if a file is selected, try
+	 * to fill in the edge textfield as well.
+	 */
+	@FXML
+	private void onNodeFieldClicked(MouseEvent e) {
+		File nodeFile = selectFile("Node file", NODE);
+		if (nodeFile != null) {
+			graphLoadService.setNodeFile(nodeFile);
+			nodeField.setText(graphLoadService.getNodeFile().getAbsolutePath());
+			graphLoadService.setEdgeFile(openEdgeFile(nodeFile.getPath()));
+			edgeField.setText(graphLoadService.getEdgeFile().getAbsolutePath());
+		}
+	}
+
+	/*
+     * If the edge textfield is clicked, open the filechooser and if a file is selected, try
+     * to fill in the node textfield as well.
+     */
+	@FXML
+	private void onEdgeFieldClicked(Event e) {
+		File edgeFile = selectFile("Edge file", EDGE);
+		if (edgeFile != null) {
+			graphLoadService.setEdgeFile(edgeFile);
+			edgeField.setText(graphLoadService.getEdgeFile().getAbsolutePath());
+			graphLoadService.setNodeFile(openNodeFile(edgeFile.getPath()));
+			nodeField.setText(graphLoadService.getNodeFile().getAbsolutePath());
+		}
+	}
+
+	/*
+	 * If the newick textfield is clicked, open the filechooser and if a file is selected,
+	 * fill in the newick textfield.
+	 */
+	@FXML
+	private void onNewickFieldClicked(Event e) {
+		File newickFile = selectFile("Newick file", NEWICK);
+		if (newickFile != null) {
+			newickLoadService.setNewickFile(newickFile);
+			newickField.setText(newickLoadService.getNewickFile().getAbsolutePath());
+		}
+	}
+
+	/*
+	 * If the open button is clicked,
+	 */
+	@FXML
+	private void onOpenAction(ActionEvent actionEvent) {
+		/* Can't be created in initialize() since fileOpenPane is still 0 at that point */
```

Comment:
I did test this. Can change it so it checks if it is null and if so, create it. However, progress bars are not meant to be reused imo because it needs resetting after it is done.